### PR TITLE
[gdal] Update to 3.4.3

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Updates GDAL to 3.4.3.
  Makes atlmfc dependency opt-in. Fixes mingw builds. Closes #24010. 
  Improves patches.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Upstream release candidate. I am still working on this PR.

### CODE REVIEW

File: ports/gdal/0006-Fix-mingw-dllexport.patch [hunk 1]
```diff
@@ -7,7 +7,7 @@ index 98805cf..a6a1846 100644
  
  #ifndef CPL_DLL
 -#if defined(_MSC_VER) && !defined(CPL_DISABLE_DLL)
-+#if defined(WIN32) && !defined(CPL_DISABLE_DLL)
++#if defined(WIN32) && (!defined(CPL_DISABLE_DLL) || defined(DLL_EXPORT))
```
Comment (dg0yt) [inline ports/gdal/0006-Fix-mingw-dllexport.patch:10]: `_WIN32` ?!

Comment (Cheney-W) [reply ports/gdal/0006-Fix-mingw-dllexport.patch:10]: It should be WIN32.
Upstream have following [contents](https://github.com/OSGeo/gdal/blob/master/port/cpl_port.h#:~:text=/*%20%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%20*/-,%23if%20defined(_WIN32)%20%26%26%20!defined(WIN32),%23endif,-/*%20%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%20*/):
```
/* ==================================================================== */
/*      We will use WIN32 as a standard windows define.                 */
/* ==================================================================== */
#if defined(_WIN32) && !defined(WIN32)
#  define WIN32
#endif

#if defined(_WINDOWS) && !defined(WIN32)
#  define WIN32
#endif
```

Comment (dg0yt) [reply ports/gdal/0006-Fix-mingw-dllexport.patch:10]: So the PR is complete.

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**